bpo-35995: update logging.handlers.SMTPHandler to support smtp server which just in TLS mode

lidayan opened the pull request:
[bpo-35995](https://bugs.python.org/issue35995): Fix some bug in logging module

just setting param secure=[] means use ssl socket with default certfile
but smtplib.SMTP() failed in self.connect --> self._get_socket. because it is not SMTP_SSL._get_socket

<!-- issue-number: [bpo-35995](https://bugs.python.org/issue35995) -->
https://bugs.python.org/issue35995
<!-- /issue-number -->

Commit:
bpo-35995: Fix some bug in logging module

setting secure=[] but smtplib.SMTP() failed in self.connect --> self._get_socket. because it is not SMTP_SSL._get_socket

Commit:
bpo-35995: Fix some bug in logging module

setting secure=[] but smtplib.SMTP() failed in self.connect --> self._get_socket. because it is not SMTP_SSL._get_socket

Commit:
bpo-35995: Fix some bug in logging module just format code

setting secure=[] but smtplib.SMTP() failed in self.connect --> self._get_socket. because it is not SMTP_SSL._get_socket

lidayan commented on Lib/logging/handlers.py:
```diff
@@ -1053,7 +1056,7 @@ def __init__(self, appname, dllname=None, logtype="Application"):
                 logging.ERROR   : win32evtlog.EVENTLOG_ERROR_TYPE,
                 logging.CRITICAL: win32evtlog.EVENTLOG_ERROR_TYPE,
-         }
+            }
```
format code style by hande

eamanu reviewed (changes requested):
Please, could you add some test? Also, I think that this need have a news. 

And please sign CLA

JulienPalard reviewed (changes requested):
Hi, thanks for your contribution!

Looks like you're changing from explicit TLS (STARTSSL) to implicit TLS, this may fix your communication with your server, but this will break the communcation to servers using explicit TLS.

Commit:
bpo-35995: Fix some bug in logging module just format code

setting secure=[] but smtplib.SMTP() failed in self.connect --> self._get_socket. because it is not SMTP_SSL._get_socket

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

Commit:
add new logHandler SMTPSSLHandler instead of updating SMTPHandler

Commit:
Merge branch 'fix-issue-35995' of https://github.com/lidayan/cpython into fix-issue-35995

Commit:
add new logHandler SMTPSSLHandler instead of updating SMTPHandler

Commit:
Revert "add new logHandler SMTPSSLHandler instead of updating SMTPHandler"

This reverts commit 5778cc5f

Commit:
Revert "add new logHandler SMTPSSLHandler instead of updating SMTPHandler"

This reverts commit b97ba417

Commit:
update logging.handlers.SMTPHandler to support smtp server which just in TLS mode
